assets/js/editor/editor.js

```diff
@@ -614,19 +621,87 @@ class Communicator {
     };
     const response = await fetch(SOLUTIONS_EDITOR_URL, requestConfig);
     if (response.status !== 200) {
-      alert(getString(msgs.error_saving_files));
+      showAlert(getString(msgs.error_saving_files));
       return;
     }
     const data = await response.json();
     window.location.assign(data.success ? SOLUTIONS_LIST_URL : SOLUTIONS_EDITOR_URL);
   }
+
+  async checkSyntax() {
+    if (!SYNTAX_CHECK_URL) return;
+    const payload = {
+      files: fileBuilder.files.map((file) => {
+        return { name: file.fileName, contents: file.fileContent };
+      }),
+    };
+    const requestConfig = {
+      method: "POST",
+      headers: {
+        "X-CSRFToken": CSRF_TOKEN,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(payload),
+    };
+    const response = await fetch(SYNTAX_CHECK_URL, requestConfig);
+    if (response.status !== 200) {
+      showAlert(getString(msgs.error_checking_syntax));
+      return;
+    }
+    return await response.json();
+  }
+}
+
+class SyntaxCheckConsole {
+  constructor() {
+    this.consoleContainerElement = document.getElementById(CONSOLE_CONTAINER_ID);
+    this.consoleContentElement = document.getElementById(CONSOLE_CONTENT_ID);
+    this.consoleHideBtn = document.getElementById(CONSOLE_HIDE_BUTTON_ID);
+    this.consoleHideBtn.addEventListener("click", () => this.show(false));
+  }
+
+  show(show) {
+    this.consoleContainerElement.style.display = show ? "block" : "none";
+    tabBar.editor.editor.resize();
+  }
+
+  createOutputElement(err) {
+    const s = (text) => `<strong>${text}</strong>`;
+    const uc = (text) => `${text[0].toUpperCase()}${text.slice(1)}`; 
+    const p = document.createElement("p");
+    let className;
+    if (err.type == "error") {
+      className = "error";
+    } else if (err.type == "warning") {
+      className = "warning";
+    }
+    p.innerHTML = `${s(uc(err.type)) } in line ${s(err.line)} of ${s(err.file)}: ${err.message}`;
+    className && (p.className = `console-content--${className}`);
+    return p;
+  }
+
+  setContent(checkResult) {
+    if (checkResult.success) {
+      this.consoleContentElement.innerHTML = `<p>${getString(msgs.syntax_check_successful)}`;
```

Comment:
```suggestion
      this.consoleContentElement.innerHTML = `<p>${getString(msgs.syntax_check_successful)}</p>`;
```
Otherwise, the `innerHTML` issue applies again.